packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/ui_components/pin_code_component.dart

```diff
@@ -207,30 +186,47 @@ class _PinCodeComponentState extends State<PinCodeComponent>
             builder: (context, areBiometricsEnabled, bloc) =>
                 RxBlocBuilder<PinCodeBlocType, List<BiometricsAuthType>>(
               state: (bloc) => bloc.states.availableBiometrics,
-              builder: (context, availableBiometrics, bloc) =>
-                  _buildPageContent(
-                isPinCodeIsSecureStorage: isPinCodeInSecureStorage.hasData &&
-                    isPinCodeInSecureStorage.data!,
-                context: context,
-                hasFingerScan: areBiometricsEnabled.hasData &&
-                    areBiometricsEnabled.data! &&
-                    availableBiometrics.hasData &&
-                    availableBiometrics.data!
-                        .contains(BiometricsAuthType.fingerprint),
-                hasFaceScan: areBiometricsEnabled.hasData &&
-                    areBiometricsEnabled.data! &&
-                    availableBiometrics.hasData &&
-                    availableBiometrics.data!.contains(BiometricsAuthType.face),
-                biometricsEnabled: areBiometricsEnabled.hasData
-                    ? areBiometricsEnabled.data!
-                    : false,
+              builder: (context, availableBiometrics, bloc) => Container(
+                decoration: BoxDecoration(
+                  gradient: LinearGradient(
+                    begin: Alignment.topCenter,
+                    end: Alignment.bottomCenter,
+                    colors: [
+                      context.pinCodeTheme.primaryGradientStart,
+                      context.pinCodeTheme.primaryGradientEnd
+                    ],
+                  ),
+                ),
+                padding: EdgeInsets.symmetric(
+                    horizontal: MediaQuery.of(context).size.width * 0.1,
+                    vertical: 20),
+                child: _buildPageContent(
+                  pinLength: widget.pinLength,
+                  isPinCodeIsSecureStorage: isPinCodeInSecureStorage.hasData &&
+                      isPinCodeInSecureStorage.data!,
+                  context: context,
+                  hasFingerScan: areBiometricsEnabled.hasData &&
```

Comment:
Complex conditions can be extracted as a private methods to reduce the code build method as well as making it more readable